### PR TITLE
add weibull to util

### PR DESCRIPTION
**Pull request by jq**
@deerzq @hectorgon

### CODE REVIEW

File: core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    if (t > 0) {
+      return Math.min(Math.pow(t, 1.0/k), maxRatio);
+    } else {
+      return 0.0;
+    }
+  }
+
+  // return function = x*p(x) where p(x) = e^(ax^k + b)
+  // derivative of return = (a*kx^k+1)*e^(ax^k)*e^b
+  public double suggestRatioMaxRevenueP1() {
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java]: I feel this is too pricing specific ......

File: core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package com.airbnb.aerosolve.core.util;
+
+import lombok.experimental.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+/*
+  Prob(p) = exp(a*(p/base)^k + b)
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java]: what is `base`?

File: core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  p is between 0 and maxRatio*base
+  so the formula can also be:
+  Prob(ratio) = exp(a*(ratio)^k + b)
+ */
+@Slf4j @Builder public class Weibull {
+  final private double k, a, b, maxRatio;
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java]: remove `maxRatio`

File: core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  public double getProbabilityForRatio(double ratio) {
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java]: rename input as `x`

File: core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    return Math.exp(a * Math.pow(ratio, k) + b);
+  }
+
+  public double suggestRatioWithTargetProbability(double targetProbability) {
```
Comment (deerzq) [inline core/src/main/java/com/airbnb/aerosolve/core/util/Weibull.java]: perhaps remove all pricing specific methods.

**Comment by jq** [comment]
@deerzq  Thanks for review!! updated with general naming.

**Comment by deerzq** [comment]
LGTM

**Pull request merged**